internal/services/worker.go

```diff
@@ -0,0 +1,55 @@
+package services
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/celestiaorg/talis/internal/logger"
+)
+
+// LaunchWorker launches a goroutine that will initialize the worker and execute tasks
+func LaunchWorker(ctx context.Context, wg *sync.WaitGroup, taskService *Task) {
+	defer wg.Done()
+	const taskLimit = 10
+	const backoff = time.Second
+
+	logger.Info("Worker started")
+
+	for {
+		select {
+		case <-ctx.Done():
+			logger.Info("Worker received shutdown signal, stopping...")
+			return
+		default:
+		}
+
+		// Fetch schedulable tasks
+		tasks, err := taskService.GetSchedulableTasks(ctx, taskLimit)
+		if err != nil {
+			logger.Errorf("Worker error fetching tasks: %v", err)
+			// Wait before retrying to avoid spamming logs on persistent DB errors
+			time.Sleep(backoff)
```

Comment:
instead of all these `time.Sleep`s , a `ticker` could be used in the `for`

Reply:
I thought about it, initially I didn't do it because it didn't reduce the code complexity, but I'm realizing now the more important benefit is blocking at the select statement to also here the shutdown signal if that came in. 

Will update